lib/set-archive-author.php

```diff
@@ -0,0 +1,28 @@
+<?php
+
+// Find a username specified in the option "archive_author"
+// Set the post_author (in $data) to be the id corresponding to the user
+
+function set_archive_author($data, $postarr)
+{
+	$fix_types = ["page", "post", "attachment"];
+
+	if (!in_array($data['post_type'], $fix_types)) {
+		return $data;
+	}
+
+	$archive_username = get_option('archive_author');
```

Comment:
Bug: it looks like this should be `$archive_author = get_option('archive_author');`. The `$archive_username` variable isn't used again here.